bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -267,6 +275,36 @@ public void getURLInput(String url) throws IOException {
         }
     }
 
+    /**
+     * Builds the {@link TimeSeries} that represents the future tibber prices.
+     *
+     * @param today The prices for today
+     * @param tomorrow The prices for tomorrow.
+     * @return The {@link TimeSeries} with future values.
+     */
+    private TimeSeries buildTimeSeries(JsonArray today, JsonArray tomorrow) {
+        final TimeSeries timeSeries = new TimeSeries(TimeSeries.Policy.REPLACE);
+        mapTimeSeriesEntries(today, timeSeries);
+        mapTimeSeriesEntries(tomorrow, timeSeries);
+        return timeSeries;
+    }
+
+    private static void mapTimeSeriesEntries(JsonArray today, TimeSeries timeSeries) {
+        for (JsonElement entry : today) {
+            final Instant startsAt = parseDateTime(entry.getAsJsonObject().get("startsAt"));
+            final Instant endsAt = startsAt.plus(1, ChronoUnit.HOURS).minus(1, ChronoUnit.MICROS);
+            final DecimalType value = new DecimalType(entry.getAsJsonObject().get("total").getAsString());
+            // As the value is valid for exactly one hour and is not linear, add one entry at the beginning
+            // and one at the end of the interval, so it's a step function instead a linear one.
+            timeSeries.add(startsAt, value);
+            timeSeries.add(endsAt, value);
+        }
+    }
+
+    private static Instant parseDateTime(JsonElement element) {
+        return ZonedDateTime.parse(element.getAsString()).toInstant();
```

Comment:
This can throw [DateTimeParseException](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/format/DateTimeParseException.html), which should be caught so that the binding will not crash on unexpected JSON data.

Reply:
done.